plugins/poll/app/serializers/poll_serializer.rb

```diff
@@ -42,7 +42,13 @@ def include_groups?
   end
 
   def options
-    object.poll_options.map { |o| PollOptionSerializer.new(o, root: false).as_json }
+    object.poll_options.map do |option|
+      PollOptionSerializer.new(
+        option,
+        root: false,
+        scope: { can_see_results: object.can_see_results?(scope.user) }
```

Comment:
Maybe we could check `object.can_see_results?(scope.user)` once out of the loop and use that result in the loop?

Reply:
:+1: implemented in last commit